src/openqaoa-core/openqaoa/algorithms/qaoa/qaoa_result.py

```diff
@@ -629,3 +629,56 @@ def lowest_cost_bitstrings(self, n_bitstrings: int = 1) -> dict:
             ],
         }
         return best_results
+    
+    def calculate_statistics(self, include_intermediate=False) -> dict:
+        """
+        A function to calculate statistics of measurement outcomes associated with a QAOA workflow
+
+        Parameters
+        ----------
+        include_intermediate: `bool`
+            Whether it is necessary to calculate statistics for intermediate results. Defaults to False.
+        """
+        if len(self.intermediate["measurement_outcomes"]) == 0 and include_intermediate == True:
+            raise ValueError(
+                "Please specify saving intermediate measurements during optimization."
+            )
+
+        if isinstance(self.optimized["measurement_outcomes"], dict):
+            optimized_measurement_outcomes = self.optimized["measurement_outcomes"]
+        elif isinstance(self.optimized["measurement_outcomes"], np.ndarray):
+            optimized_measurement_outcomes = self.get_counts(
+                self.optimized["measurement_outcomes"]
+            )
+        else:
+            raise TypeError(
+                f"The measurement outcome {type(self.optimized['measurement_outcomes'])} is not valid."
+            )
+
+        if isinstance(self.intermediate["measurement_outcomes"], list):
+            if len(self.intermediate["measurement_outcomes"]) > 0:
+                if isinstance(self.intermediate["measurement_outcomes"][0], dict):
+                    intermediate_measurement_outcomes = self.intermediate["measurement_outcomes"]
+                elif isinstance(self.intermediate["measurement_outcomes"][0], np.ndarray):
+                    intermediate_measurement_outcomes = [self.get_counts(i) for i in self.intermediate["measurement_outcomes"]]
+                else:
+                    raise TypeError(
+                        f"The measurement outcome {type(self.intermediate['measurement_outcomes'][0])} is not valid."
+                    )
+        else:
+            raise TypeError(
+                f"The measurement outcome {type(self.intermediate['measurement_outcomes'])} is not valid."
+            )
+
+        def sorted_mean_std_deviation(counts: dict):
+            values = list(counts.values())
+            return {
+                'sorted': dict(sorted(counts.items(), key=lambda x: x[1], reverse=True)),
```

Comment:
I am having second thoughts about this, perhaps we can remove it?

I am thinking it may not be very informative after all, and the user can always compute it if they so wish.

On the other hand, maybe we could return also mean and std for some percentiles? What do you think?

Reply:
Actually, no need to implement this now. I think it is fine the way it is :)